### ISSUE TITLE
How break three columns on two col an 1 col in center by mjml group?

### ISSUE THREAD

**Issue by Hodakhosravi** [mjmlio/mjml#2326]
Hi,
My email has 3 columns in a section.
The elements are 3 social contacts. I separated them into 2 columns to let them side by side and one column in the next line in mobile (I used mjml group).
But, on mobile, the last column is not centered. 
  for example, if I have 5 elements, I want to let 3 on a line and 2 on a second line. 
and another issue. When I use mjml table to align an image with a text in the column it doesn't center too.

**Comment by Hodakhosravi** [comment]
this is related to my second issue:

`<mjml>
  <mj-body width="600px">
    
 <mj-section background-url="https://mcmaster.onlinefiles.ca/test_images/CTA_Block_bkg.png"  background-size="cover"  background-repeat="no-repeat" padding="20px" >

<mj-column  width="70%"  padding="10px">
  <mj-table >
           <tr>
             <td >
               <img src="https://mcmaster.onlinefiles.ca/test_images/Icon_Phone.png" width="70%" padding="0px"/>
             </td>
             <td >
               <img src="https://mcmaster.onlinefiles.ca/test_images/WhatsApp_BTN.png" width="60%" />
             </td>
           </tr>
</mj-table>
</mj-column>

</mj-section>
  </mj-body>
</mjml>`

**Comment by iRyusa** [comment]
Hi, 

Your second issue seems to render fines here https://mjml.io/try-it-live/HQ6NNCkDK0r can you provide more details about it (on which clients it appears, mjml version used, how did you send the email) ?

I don't really understand the first one either, can you share a sample code as you've done for the first issue please ?

**Comment by Hodakhosravi** [comment]
![Untitled](https://user-images.githubusercontent.com/74387098/125285346-d9f06f80-e31a-11eb-9ca7-c38a217a4cb6.png)

**Comment by Hodakhosravi** [comment]
> Hi,
> 
> Your second issue seems to render fines here https://mjml.io/try-it-live/HQ6NNCkDK0r can you provide more details about it (on which clients it appears, mjml version used, how did you send the email) ?
> 
> I don't really understand the first one either, can you share a sample code as you've done for the first issue please ?
`<mjml>
  <mj-head>
    <mj-title>McMaster-Image Cards</mj-title>
    <mj-attributes>
      
      <link rel="preconnect" href="https://fonts.googleapis.com">
      <link rel="preconnect" href="https://fonts.gstatic.com">
      <link href="https://fonts.googleapis.com/css2?family=Montserrat:wght@500&display=swap" rel="stylesheet">
    </mj-attributes>

    
        <mj-style>
        @media (max-width:480px) {
            .center {
              text-align: center;
            }
          }
        .link {
         color: #ffffff;
         text-decoration:none;
        }
        .link-gray {
         color: #5E6A71;
         text-decoration:none;
        }
        .shadow {
          -webkit-box-shadow:1px 1px 12px 5px rgb(219 219 221);
          -moz-box-shadow:1px 1px 12px 5px rgb(219 219 221);
          box-shadow: 1px 1px 12px 5px rgb(219 219 221);
          /* box-shadow: 2px 2px 2px  rgb(136, 136, 136); */
         }
        </mj-style>
  </mj-head>
  
  <mj-body background-color="#F3F1F1" >
    <mj-section>
          <mj-column width="100%" background-color="#ffffff" padding="0px 1px" >
      <mj-text color="#5E6A71" align="center" font-weight="bold" font-size="14px" line-height="25px" font-family= "Montserrat, sans-serif" >
        Image Cards 1
      </mj-text>
    </mj-column>
    </mj-section>
    
    <mj-section background-color="#ffffff" padding="20px 0px 5px 0px" >

      <mj-column background-color="#ffffff" padding="0px" width="100%" >
        <mj-text color="black" align="center" font-weight="400" font-size="25px" line-height="22px" font-family= "Montserrat, sans-serif" >
    <b>About McMaster</b>
        </mj-text>
    </mj-column>

    </mj-section>
<mj-section background-color="#ffffff" padding="20px" >
  <mj-column background-color="#ffffff" padding="0px" width="40%" css-class="shadow" >
      <mj-image src="https://mcmaster.onlinefiles.ca/test_images/Card_1.png"  padding="0px"> </mj-image>
    <mj-text color="black" align="left" font-weight="400" font-size="15px" line-height="22px" font-family= "Montserrat, sans-serif" >
<b>sit amet lacus commodo maximus ut eu arcu.</b>
    </mj-text>
  </mj-column>
  <mj-column  width="5%">
    
   <mj-text color="#ffffff" align="center" >
      _______
      </mj-text>
    </mj-column>
  <mj-column background-color="#ffffff" padding="0px" width="40%"  css-class="shadow"  >
      <mj-image src="https://mcmaster.onlinefiles.ca/test_images/Card_2.png"  padding="0px"> </mj-image>
    <mj-text color="black" align="left" font-weight="400" font-size="15px" line-height="22px" font-family= "Montserrat, sans-serif" >
<b>sit amet lacus commodo maximus ut eu arcu.</b>
    </mj-text>
  </mj-column>
</mj-section>




<mj-section>
      <mj-column width="100%" background-color="#ffffff" padding="0px 1px" >
  <mj-text color="#5E6A71" align="center" font-weight="bold" font-size="14px" line-height="25px" font-family= "Montserrat, sans-serif" >
    Image Cards 2
  </mj-text>
</mj-column>
</mj-section>
<mj-section background-color="#ffffff" padding="20px 0px 5px 0px" >

  <mj-column background-color="#ffffff" padding="0px" width="100%" >
    <mj-text color="black" align="center" font-weight="400" font-size="25px" line-height="22px" font-family= "Montserrat, sans-serif" >
<b>About McMaster</b>
    </mj-text>
</mj-column>

</mj-section>
<mj-section background-color="#ffffff" padding="20px" >
  
<mj-column background-color="#ffffff" padding="0px" width="30%"  css-class="shadow" >
  <mj-image src="https://mcmaster.onlinefiles.ca/test_images/Card_3.png"  padding="0px"> </mj-image>
<mj-text color="black" align="left" font-weight="400" font-size="12px" line-height="18px" font-family= "Montserrat, sans-serif" >
<b>sit amet lacus commodo maximus ut eu arcu.</b>
</mj-text>
</mj-column>
<mj-column  width="2%">

<mj-text color="#ffffff" align="center" >
  _______
  </mj-text>
  </mj-column>
<mj-column background-color="#ffffff" padding="0px" width="30%"  css-class="shadow" css-class="mobilewide">
  <mj-image src="https://mcmaster.onlinefiles.ca/test_images/Card_4.png"  padding="0px"> </mj-image>
<mj-text color="black" align="left" font-weight="400" font-size="12px" line-height="18px"  font-family= "Montserrat, sans-serif" >
<b>sit amet lacus commodo maximus ut eu arcu.</b>
</mj-text>
</mj-column>
<mj-column  width="2%">

<mj-text color="#ffffff" align="center" >
  _______
  </mj-text>
</mj-column>
<mj-column background-color="#ffffff" padding="0px" width="30%"   css-class="shadow"  >
  <mj-image src="https://mcmaster.onlinefiles.ca/test_images/Card_5.png"  padding="0px"> </mj-image>
<mj-text color="black" align="left" font-weight="400" font-size="12px" line-height="18px"  font-family= "Montserrat, sans-serif" >
<b>sit amet lacus commodo maximus ut eu arcu.</b>
</mj-text>
</mj-column>
</mj-section>

</mj-body>
</mjml>`

**Comment by iRyusa** [comment]
Really hard to follow you here you don't use group in your exemple code 🤔 

Can you maybe add some screenshot on what you're trying to do and the current behavior with your mj-groups ? 

Columns extends to 100% of content in mobile so it's most likely why you think it's not centered.

This looks perfectly fine in the try it live 
![image](https://user-images.githubusercontent.com/570317/125286667-5e8fbd80-e31c-11eb-97f3-748d0d841b5b.png)

**Comment by Hodakhosravi** [comment]
I didn't add mjml group because it is not working. I will attach how it rendered when I add group. about the second issue as you see it has huge padding between images

**Comment by Hodakhosravi** [comment]
![Untitled](https://user-images.githubusercontent.com/74387098/125288798-d3fc8d80-e31e-11eb-8398-34a494d558ba.png)

**Comment by Hodakhosravi** [comment]
`<mjml>
  <mj-head>
    <mj-title>McMaster-Image Cards</mj-title>
    <mj-attributes>
      
      <link rel="preconnect" href="https://fonts.googleapis.com">
      <link rel="preconnect" href="https://fonts.gstatic.com">
      <link href="https://fonts.googleapis.com/css2?family=Montserrat:wght@500&display=swap" rel="stylesheet">
    </mj-attributes>

    
        <mj-style>
        @media (max-width:480px) {
            .center {
              text-align: center;
            }
          }
        .link {
         color: #ffffff;
         text-decoration:none;
        }
        .link-gray {
         color: #5E6A71;
         text-decoration:none;
        }
        .shadow {
          -webkit-box-shadow:1px 1px 12px 5px rgb(219 219 221);
          -moz-box-shadow:1px 1px 12px 5px rgb(219 219 221);
          box-shadow: 1px 1px 12px 5px rgb(219 219 221);
          /* box-shadow: 2px 2px 2px  rgb(136, 136, 136); */
         }
        </mj-style>
  </mj-head>
  
  <mj-body background-color="#F3F1F1" >
    <mj-section>
          <mj-column width="100%" background-color="#ffffff" padding="0px 1px" >
      <mj-text color="#5E6A71" align="center" font-weight="bold" font-size="14px" line-height="25px" font-family= "Montserrat, sans-serif" >
        Image Cards 1
      </mj-text>
    </mj-column>
    </mj-section>
    
    <mj-section background-color="#ffffff" padding="20px 0px 5px 0px" >

      <mj-column background-color="#ffffff" padding="0px" width="100%" >
        <mj-text color="black" align="center" font-weight="400" font-size="25px" line-height="22px" font-family= "Montserrat, sans-serif" >
    <b>About McMaster</b>
        </mj-text>
    </mj-column>

    </mj-section>
<mj-section background-color="#ffffff" padding="20px" >
  <mj-column background-color="#ffffff" padding="0px" width="40%" css-class="shadow" >
      <mj-image src="https://mcmaster.onlinefiles.ca/test_images/Card_1.png"  padding="0px"> </mj-image>
    <mj-text color="black" align="left" font-weight="400" font-size="15px" line-height="22px" font-family= "Montserrat, sans-serif" >
<b>sit amet lacus commodo maximus ut eu arcu.</b>
    </mj-text>
  </mj-column>
  <mj-column  width="5%">
    
   <mj-text color="#ffffff" align="center" >
      _______
      </mj-text>
    </mj-column>
  <mj-column background-color="#ffffff" padding="0px" width="40%"  css-class="shadow"  >
      <mj-image src="https://mcmaster.onlinefiles.ca/test_images/Card_2.png"  padding="0px"> </mj-image>
    <mj-text color="black" align="left" font-weight="400" font-size="15px" line-height="22px" font-family= "Montserrat, sans-serif" >
<b>sit amet lacus commodo maximus ut eu arcu.</b>
    </mj-text>
  </mj-column>
</mj-section>




<mj-section>
      <mj-column width="100%" background-color="#ffffff" padding="0px 1px" >
  <mj-text color="#5E6A71" align="center" font-weight="bold" font-size="14px" line-height="25px" font-family= "Montserrat, sans-serif" >
    Image Cards 2
  </mj-text>
</mj-column>
</mj-section>
<mj-section background-color="#ffffff" padding="20px 0px 5px 0px" >

  <mj-column background-color="#ffffff" padding="0px" width="100%" >
    <mj-text color="black" align="center" font-weight="400" font-size="25px" line-height="22px" font-family= "Montserrat, sans-serif" >
<b>About McMaster</b>
    </mj-text>
</mj-column>

</mj-section>
<mj-section background-color="#ffffff" padding="20px" >
  <mj-group>
<mj-column background-color="#ffffff" padding="0px" width="30%"  css-class="shadow" >
  <mj-image src="https://mcmaster.onlinefiles.ca/test_images/Card_3.png"  padding="0px"> </mj-image>
<mj-text color="black" align="left" font-weight="400" font-size="12px" line-height="18px" font-family= "Montserrat, sans-serif" >
<b>sit amet lacus commodo maximus ut eu arcu.</b>
</mj-text>
</mj-column>

<mj-column  width="2%">

<mj-text color="#ffffff" align="center" >
  _______
  </mj-text>
  </mj-column>
  
<mj-column background-color="#ffffff" padding="0px" width="30%"  css-class="shadow" >
  <mj-image src="https://mcmaster.onlinefiles.ca/test_images/Card_4.png"  padding="0px"> </mj-image>
<mj-text color="black" align="left" font-weight="400" font-size="12px" line-height="18px"  font-family= "Montserrat, sans-serif" >
<b>sit amet lacus commodo maximus ut eu arcu.</b>
</mj-text>
</mj-column>
      </mj-group>
<mj-column  width="2%">

<mj-text color="#ffffff" align="center" >
  _______
  </mj-text>
</mj-column>
<mj-column background-color="#ffffff" padding="0px" width="30%"   css-class="shadow"  >
  <mj-image src="https://mcmaster.onlinefiles.ca/test_images/Card_5.png"  padding="0px"> </mj-image>
<mj-text color="black" align="left" font-weight="400" font-size="12px" line-height="18px"  font-family= "Montserrat, sans-serif" >
<b>sit amet lacus commodo maximus ut eu arcu.</b>
</mj-text>
</mj-column>
</mj-section>

</mj-body>
</mjml>`

**Comment by Hodakhosravi** [comment]
I think I should hide the padding column in mobile view

**Comment by iRyusa** [comment]
You need to play with group width and column width inside it https://mjml.io/try-it-live/uk9fSFUK7Fn
Column size are relative to parent group width.

**Comment by Hodakhosravi** [comment]
Thanks, I didn't know that we can play with group width too.

**Comment by Hodakhosravi** [comment]
and it looks when I group 2 columns together. in desktop view
![Untitled](https://user-images.githubusercontent.com/74387098/125290846-060eef00-e321-11eb-8a26-fc756b333b61.png)

**Comment by Hodakhosravi** [comment]
finally, I came out with this 👍🏻 
![Untitled](https://user-images.githubusercontent.com/74387098/125345233-91f03d80-e358-11eb-8cf3-6f58e770a810.png)
